### PR TITLE
Fix issue where readonly partition lacked unique columns constraint

### PR DESCRIPTION
**Pull request by achocron**
Things to consider before you submit the PR:

* [ ] Has `CHANGELOG.md` been updated?
* [x] Are tests passing locally?
* [x] Are the files formatted correctly?
* [x] Did you add unit tests?
* [ ] Did you check UI tests on the sample app? They are not executed on CI.
* [ ] Did you test your change with either the sample apps that are included in the repository or with a blank app that uses your change?

## Description
The readonly storage table was not being created with the unique columns constraint that all tables require. As a result, when a document in the readonly table was replaced (usually to update TTL), it was actually duplicated instead of replaced.

### CODE REVIEW

**Comment by yochaude** [comment]
What was missing for the upsert not to work? I cannot really tell from this PR.

File: AppCenter/AppCenterTests/MSLogDBStorageTests.m [hunk 1]
```diff
@@ -1061,13 +1061,20 @@ - (void)testErrorDeletingOldestLog {
...
+  [self.storageTestUtil deleteDatabase];
+
   // When
   self.sut = [MSLogDBStorage new];
 
   // Then
   NSString *currentTable =
       [self.sut executeSelectionQuery:[NSString stringWithFormat:@"SELECT sql FROM sqlite_master WHERE name='%@'", kMSLogTableName]][0][0];
   assertThat(currentTable, is(kMSLatestSchema));
+//  NSArray *result =
```
Comment (jwallra) [inline AppCenter/AppCenterTests/MSLogDBStorageTests.m]: Remove commented code... Actually why were we saving the result before?  It doesn't look like it's used

Comment (YiMysty) [reply AppCenter/AppCenterTests/MSLogDBStorageTests.m]: This is the test code, I will remove it!

Comment (YiMysty) [reply AppCenter/AppCenterTests/MSLogDBStorageTests.m]: removed! :D

File: AppCenter/AppCenterTests/MSLogDBStorageTests.m [hunk 2]
```diff
@@ -1061,6 +1061,9 @@ - (void)testErrorDeletingOldestLog {
 
 - (void)testCreateFromLatestSchema {
 
+  // If
+  [self.storageTestUtil deleteDatabase];
```
Comment (dhei) [inline AppCenter/AppCenterTests/MSLogDBStorageTests.m]: `[self.storageTestUtil deleteDatabase]` is already called in [`(void)tearDown`](https://github.com/Microsoft/AppCenter-SDK-Apple/pull/1499/files#diff-ae42d5f18ad4a9fc0a4426cbbb33eb98R57), why duplicate here?

File: AppCenterDataStorage/AppCenterDataStorageTests/MSDBDocumentStoreTests.m [hunk 3]
```diff
@@ -202,6 +200,29 @@ - (void)testReadUserDocumentFromLocalDatabaseNotFound {
   XCTAssertEqualObjects(documentWrapper.documentId, documentId);
 }
 
+- (void)testUpsertReplacesCorrectlyInAppStorage {
```
Comment (dhei) [inline AppCenterDataStorage/AppCenterDataStorageTests/MSDBDocumentStoreTests.m]: Test case suggestion:

On Android we have a test case (https://github.com/Microsoft/AppCenter-SDK-Android/blob/b563fc68e6919d2a20dd12fd367ca4c080b1f7f9/sdk/appcenter/src/androidTest/java/com/microsoft/appcenter/utils/storage/DatabaseManagerAndroidTest.java#L498) that verifies `replace` called twice with insertion the first time and update the second time. 

I haven't seen any other test methods in this class verifying `replace` called only once, so maybe we do the same test as Android? 😄

File: AppCenter/AppCenter/Internals/Storage/MSDBStorage.m [hunk 4]
```diff
@@ -11,21 +11,48 @@ @implementation MSDBStorage
...
       _pageSize = [MSDBStorage getPageSizeInOpenedDatabase:db];
+      NSUInteger databaseVersion = [MSDBStorage versionInOpenedDatabase:db];
+
+      // Create table
+      [MSDBStorage createTablesWithSchema:schema inOpenedDatabase:db];
+      if (newDatabase) {
+        MSLogInfo([MSAppCenter logTag], @"Created \"%@\" database with %lu version.", filename, (unsigned long)version);
+        [self customizeDatabase:db];
+      } else if (databaseVersion < version) {
+        MSLogInfo([MSAppCenter logTag], @"Migrate \"%@\" database from %lu to %lu version.", filename, (unsigned long)databaseVersion,
```
Comment (jwallra) [inline AppCenter/AppCenter/Internals/Storage/MSDBStorage.m]:
```suggestion
        MSLogInfo([MSAppCenter logTag], @"Migrating \"%@\" database from version %lu to %lu.", filename, (unsigned long)databaseVersion,
```

File: AppCenterDataStorage/AppCenterDataStorageTests/MSDBDocumentStoreTests.m [hunk 5]
```diff
@@ -202,6 +200,37 @@ - (void)testReadUserDocumentFromLocalDatabaseNotFound {
...
+
+  // If
+  MSDocumentWrapper *expectedDocumentWrapper = [MSDocumentUtils documentWrapperFromData:[self jsonFixture:@"validTestDocument"]
+                                                                           documentType:[MSDictionaryDocument class]];
+  MSWriteOptions *writeOptions = [[MSWriteOptions alloc] initWithDeviceTimeToLive:MSDataStoreTimeToLiveInfinite];
+
+  // When
+  // Upsert twice to ensure that replacement is correct.
```
Comment (jwallra) [inline AppCenterDataStorage/AppCenterDataStorageTests/MSDBDocumentStoreTests.m]: This is no longer upserting twice in this part, comment should be corrected

Comment (achocron) [reply AppCenterDataStorage/AppCenterDataStorageTests/MSDBDocumentStoreTests.m]: Just meant in general we do it twice, not necessarily in that block i guess

**Pull request merged**